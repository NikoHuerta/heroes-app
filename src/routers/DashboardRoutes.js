import React from 'react'
import {
    Routes,
    Route
  } from 'react-router-dom';
import { DcScreen } from '../components/dc/DcScreen';
import { HeroScreen } from '../components/heroes/HeroScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { Navbar } from '../components/ui/Navbar';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />   
            <div className="container mt-2">
                <Routes>
                    <Route exact path="/marvel" element={ <MarvelScreen /> }/>
                    <Route exact path="/hero/:heroeId/*" element={ <HeroScreen /> }/>
                    <Route exact path="/dc" element={ <DcScreen /> }/>

                    <Route path="*" element={ <MarvelScreen /> } />
                </Routes>
            </div>
        </>
    )
}
