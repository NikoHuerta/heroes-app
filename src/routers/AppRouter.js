import React from 'react'
import {
    BrowserRouter as Router,
    Routes,
    Route
  } from 'react-router-dom';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';

// import { Navbar } from '../components/ui/Navbar';



export const AppRouter = () => {
    return (
    <Router>
      <div>
        {/* <Navbar /> */}

        <Routes>
          <Route exact path="/login" element={ <LoginScreen /> }/>
          <Route exact path="/*" element={ <DashboardRoutes /> }/>
        </Routes>
        
      </div>
    </Router>
    );
}
