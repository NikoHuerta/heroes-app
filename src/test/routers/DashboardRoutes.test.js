import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/authContext";
import { DashboardRoutes } from "../../routers/DashboardRoutes";



describe('Pruebas en <DashboardRoutes />', () => {

    const contextValue = {
        user:{
            logged: true,
            name: 'Pepe',
            email: 'pepe@pepe.pepe'
        }
    };
    
    
    test('Debe de mostrarse correctamente', () => {
    
    const wrapper = mount(<AuthContext.Provider value={ contextValue }>
                            <MemoryRouter initialEntries={ ['/'] }>
                                <DashboardRoutes />
                            </MemoryRouter>
                        </AuthContext.Provider>);

    expect(wrapper).toMatchSnapshot();    
    expect(wrapper.find('.text-info').text().trim()).toBe('Pepe');

    });

    test('Debe de mostrarse correctamente DC', () => {
    
        const wrapper = mount(<AuthContext.Provider value={ contextValue }>
                                <MemoryRouter initialEntries={ ['/dc'] }>
                                    <DashboardRoutes />
                                </MemoryRouter>
                            </AuthContext.Provider>);
    
        expect(wrapper).toMatchSnapshot();    
        expect(wrapper.find('.text-info').text().trim()).toBe('Pepe');
        expect(wrapper.find('h1').text().trim()).toBe('DC Screen');

    
    });

    test('Debe de mostrarse correctamente Marvel', () => {
    
        const wrapper = mount(<AuthContext.Provider value={ contextValue }>
                                <MemoryRouter initialEntries={ ['/marvel'] }>
                                    <DashboardRoutes />
                                </MemoryRouter>
                            </AuthContext.Provider>);
    
        expect(wrapper).toMatchSnapshot();    
        expect(wrapper.find('.text-info').text().trim()).toBe('Pepe');
        expect(wrapper.find('h1').text().trim()).toBe('Marvel Screen');
    
    });

    test('Debe de mostrarse correctamente About', () => {
    
        const wrapper = mount(<AuthContext.Provider value={ contextValue }>
                                <MemoryRouter initialEntries={ ['/about'] }>
                                    <DashboardRoutes />
                                </MemoryRouter>
                            </AuthContext.Provider>);
    
        expect(wrapper).toMatchSnapshot();    
        expect(wrapper.find('.text-info').text().trim()).toBe('Pepe');
        expect(wrapper.find('h1').text().trim()).toBe('About Screen');
    
    });

    test('Debe de mostrarse correctamente Search', () => {
    
        const wrapper = mount(<AuthContext.Provider value={ contextValue }>
                                <MemoryRouter initialEntries={ ['/search'] }>
                                    <DashboardRoutes />
                                </MemoryRouter>
                            </AuthContext.Provider>);
    
        expect(wrapper).toMatchSnapshot();    
        expect(wrapper.find('.text-info').text().trim()).toBe('Pepe');
        expect(wrapper.find('h1').text().trim()).toBe('Search Screen');
    });
    
    
});
