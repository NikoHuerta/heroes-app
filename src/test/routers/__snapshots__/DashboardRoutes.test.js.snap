// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRoutes /> Debe de mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
        <a
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Pepe
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    <div>
      <h1>
         Search Screen
      </h1>
      <hr />
      <div
        className="row"
      >
        <h4>
          Search Form
        </h4>
        <hr />
        <form
          onSubmit={[Function]}
        >
          <input
            autoComplete="off"
            className="form-control"
            name="searchString"
            onChange={[Function]}
            placeholder="Find Your Hero"
            type="text"
            value=""
          />
          <button
            className="btn m-1 btn-block btn-outline-primary"
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
      <div>
        <br />
        <h4>
          Results
        </h4>
        <hr />
        <br />
        <div
          className="alert alert-info"
        >
          Search a Hero
        </div>
        <div
          className="row row-cols-1 row-cols-md-3 g-4"
        />
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en <DashboardRoutes /> Debe de mostrarse correctamente About 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Pepe
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    <div>
      <h1>
         About Screen
      </h1>
      <hr />
      <div
        className="row"
      >
        <h4>
          Technology Used
        </h4>
        <hr />
        <span>
          npm pkg
        </span>
        <ul>
          <li>
            <a
              href="https://reactrouter.com/docs/en/v6"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span>
                react-router-dom : 6.0.2
              </span>
            </a>
          </li>
          <li>
            <a
              href="https://www.npmjs.com/package/query-string"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span>
                query-string : 7.0.1
              </span>
            </a>
          </li>
        </ul>
        <span>
          Custom Hooks
        </span>
        <ul>
          <li>
            <a
              href="https://github.com/NikoHuerta/mi-repo-react/tree/master/hooks/useForm"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span>
                useForm
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en <DashboardRoutes /> Debe de mostrarse correctamente DC 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
        <a
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Pepe
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    <div>
      <h1>
         DC Screen 
      </h1>
      <hr />
      <div
        className="row row-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <a
          className="my-card"
          href="/hero/dc-batman"
          onClick={[Function]}
        >
          <img
            alt="Batman"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/dc-batman.jpg"
          />
          <div
            className="profile-name"
          >
            Batman
          </div>
          <div
            className="profile-position"
          >
            Bruce Wayne
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    Detective Comics #27
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/dc-superman"
          onClick={[Function]}
        >
          <img
            alt="Superman"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/dc-superman.jpg"
          />
          <div
            className="profile-name"
          >
            Superman
          </div>
          <div
            className="profile-position"
          >
            Kal-El
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    Action Comics #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/dc-flash"
          onClick={[Function]}
        >
          <img
            alt="Flash"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/dc-flash.jpg"
          />
          <div
            className="profile-name"
          >
            Flash
          </div>
          <div
            className="profile-position"
          >
            Jay Garrick
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    Flash Comics #1
                  </p>
                  <p>
                    Jay Garrick, Barry Allen, Wally West, Bart Allen
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/dc-green"
          onClick={[Function]}
        >
          <img
            alt="Green Lantern"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/dc-green.jpg"
          />
          <div
            className="profile-name"
          >
            Green Lantern
          </div>
          <div
            className="profile-position"
          >
            Alan Scott
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    All-American Comics #16
                  </p>
                  <p>
                    Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/dc-arrow"
          onClick={[Function]}
        >
          <img
            alt="Green Arrow"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/dc-arrow.jpg"
          />
          <div
            className="profile-name"
          >
            Green Arrow
          </div>
          <div
            className="profile-position"
          >
            Oliver Queen
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    More Fun Comics #73
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/dc-wonder"
          onClick={[Function]}
        >
          <img
            alt="Wonder Woman"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/dc-wonder.jpg"
          />
          <div
            className="profile-name"
          >
            Wonder Woman
          </div>
          <div
            className="profile-position"
          >
            Princess Diana
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    All Star Comics #8
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/dc-martian"
          onClick={[Function]}
        >
          <img
            alt="Martian Manhunter"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/dc-martian.jpg"
          />
          <div
            className="profile-name"
          >
            Martian Manhunter
          </div>
          <div
            className="profile-position"
          >
            J'onn J'onzz
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    Detective Comics #225
                  </p>
                  <p>
                    Martian Manhunter
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/dc-robin"
          onClick={[Function]}
        >
          <img
            alt="Robin/Nightwing"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/dc-robin.jpg"
          />
          <div
            className="profile-name"
          >
            Robin/Nightwing
          </div>
          <div
            className="profile-position"
          >
            Dick Grayson
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    Detective Comics #38
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/dc-blue"
          onClick={[Function]}
        >
          <img
            alt="Blue Beetle"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/dc-blue.jpg"
          />
          <div
            className="profile-name"
          >
            Blue Beetle
          </div>
          <div
            className="profile-position"
          >
            Dan Garret
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    Mystery Men Comics #1
                  </p>
                  <p>
                    Dan Garret, Ted Kord, Jaime Reyes
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/dc-black"
          onClick={[Function]}
        >
          <img
            alt="Black Canary"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/dc-black.jpg"
          />
          <div
            className="profile-name"
          >
            Black Canary
          </div>
          <div
            className="profile-position"
          >
            Dinah Drake
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    Flash Comics #86
                  </p>
                  <p>
                    Dinah Drake, Dinah Lance
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en <DashboardRoutes /> Debe de mostrarse correctamente Marvel 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
        <a
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Pepe
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    <div>
      <h1>
         Marvel Screen 
      </h1>
      <hr />
      <div
        className="row row-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <a
          className="my-card"
          href="/hero/marvel-spider"
          onClick={[Function]}
        >
          <img
            alt="Spider Man"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/marvel-spider.jpg"
          />
          <div
            className="profile-name"
          >
            Spider Man
          </div>
          <div
            className="profile-position"
          >
            Peter Parker
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    Amazing Fantasy #15
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/marvel-captain"
          onClick={[Function]}
        >
          <img
            alt="Captain America"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/marvel-captain.jpg"
          />
          <div
            className="profile-name"
          >
            Captain America
          </div>
          <div
            className="profile-position"
          >
            Steve Rogers
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    Captain America Comics #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/marvel-iron"
          onClick={[Function]}
        >
          <img
            alt="Iron Man"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/marvel-iron.jpg"
          />
          <div
            className="profile-name"
          >
            Iron Man
          </div>
          <div
            className="profile-position"
          >
            Tony Stark
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    Tales of Suspense #39
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/marvel-thor"
          onClick={[Function]}
        >
          <img
            alt="Thor"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/marvel-thor.jpg"
          />
          <div
            className="profile-name"
          >
            Thor
          </div>
          <div
            className="profile-position"
          >
            Thor Odinson
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    Journey into Myster #83
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/marvel-hulk"
          onClick={[Function]}
        >
          <img
            alt="Hulk"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/marvel-hulk.jpg"
          />
          <div
            className="profile-name"
          >
            Hulk
          </div>
          <div
            className="profile-position"
          >
            Bruce Banner
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    The Incredible Hulk #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/marvel-wolverine"
          onClick={[Function]}
        >
          <img
            alt="Wolverine"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/marvel-wolverine.jpg"
          />
          <div
            className="profile-name"
          >
            Wolverine
          </div>
          <div
            className="profile-position"
          >
            James Howlett
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    The Incredible Hulk #180
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/marvel-daredevil"
          onClick={[Function]}
        >
          <img
            alt="Daredevil"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/marvel-daredevil.jpg"
          />
          <div
            className="profile-name"
          >
            Daredevil
          </div>
          <div
            className="profile-position"
          >
            Matthew Michael Murdock
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    Daredevil #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/marvel-hawkeye"
          onClick={[Function]}
        >
          <img
            alt="Hawkeye"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/marvel-hawkeye.jpg"
          />
          <div
            className="profile-name"
          >
            Hawkeye
          </div>
          <div
            className="profile-position"
          >
            Clinton Francis Barton
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    Tales of Suspense #57
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/marvel-cyclops"
          onClick={[Function]}
        >
          <img
            alt="Cyclops"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/marvel-cyclops.jpg"
          />
          <div
            className="profile-name"
          >
            Cyclops
          </div>
          <div
            className="profile-position"
          >
            Scott Summers
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    X-Men #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card"
          href="/hero/marvel-silver"
          onClick={[Function]}
        >
          <img
            alt="Silver Surfer"
            className="img img-responsive animate__animated animate__fadeIn"
            src="../assets/heroes/marvel-silver.jpg"
          />
          <div
            className="profile-name"
          >
            Silver Surfer
          </div>
          <div
            className="profile-position"
          >
            Norrin Radd
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera Aparación: 
                    <br />
                    The Fantastic Four #48
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en <DashboardRoutes /> Debe de mostrarse correctamente Search 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
        <a
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Pepe
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    <div>
      <h1>
         Search Screen
      </h1>
      <hr />
      <div
        className="row"
      >
        <h4>
          Search Form
        </h4>
        <hr />
        <form
          onSubmit={[Function]}
        >
          <input
            autoComplete="off"
            className="form-control"
            name="searchString"
            onChange={[Function]}
            placeholder="Find Your Hero"
            type="text"
            value=""
          />
          <button
            className="btn m-1 btn-block btn-outline-primary"
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
      <div>
        <br />
        <h4>
          Results
        </h4>
        <hr />
        <br />
        <div
          className="alert alert-info"
        >
          Search a Hero
        </div>
        <div
          className="row row-cols-1 row-cols-md-3 g-4"
        />
      </div>
    </div>
  </div>,
]
`;
